### PR TITLE
Pract07str-08 | Mihaela Hadzhikovakeva

### PR DESCRIPTION
**Pull request by mihaela0911**
Mihaela Hadzhikovakeva
6 6MI0600211

### CODE REVIEW

File: SI/Sem.07/Pract.07/Strings/Task08.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+//		if (symbols[i] == '\0')
+//		{
+//			break;
+//		}
+//		symbolsCount[symbols[i]]++;
+//	}
+//}
+void PrintWithoutRepeating(char* str)
+{
+	int symbolsCount[SIZE_SYMBOLS] = {};
```
Comment (dejansvetoslavov) [inline SI/Sem.07/Pract.07/Strings/Task08.cpp:20]: Is int needed here - it is a flag.

File: SI/Sem.07/Pract.07/Strings/Task08.cpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	//CountSymbols(str, symbolsCount);
+
+	char newStr[MAX_SIZE] = {};
+	int indexForNewStr = 0;
+
+	for (int i = 0; i < MAX_SIZE; i++)
+	{
+		if (str[i] == '\0')
+		{
```
Comment (dejansvetoslavov) [inline SI/Sem.07/Pract.07/Strings/Task08.cpp:29]: It is a matter of style, but this looks odd - `for (int i = 0; i < MAX_SIZE && str[I] != '\0'; i++)` or maybe a while loop with a counter would look better

File: SI/Sem.07/Pract.07/Strings/Task08.cpp [hunk 3]
```diff
@@ -0,0 +1,50 @@
+//solution without counting each symbol
+
+#include <iostream>
+const int MAX_SIZE = 100;
+const int SIZE_SYMBOLS = 128;
+
+//void CountSymbols(char* symbols, int* symbolsCount)
```
Comment (dejansvetoslavov) [inline SI/Sem.07/Pract.07/Strings/Task08.cpp:7]: You can remove the comments

**Review by dejansvetoslavov — APPROVED** [review]

**Comment by dejansvetoslavov** [comment]
Awesome that you reworked the solution!! 🥇

**Review by Justsvetoslavov — CHANGES_REQUESTED** [review]
Waiting for the above comments to be addressed.

**Review by Justsvetoslavov — APPROVED** [review]

**Pull request merged**